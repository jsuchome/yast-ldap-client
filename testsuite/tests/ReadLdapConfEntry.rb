# encoding: utf-8

#  ReadLdapConfEntry.ycp
#  Test of Ldap::ReadLdapConfEntry function
#  Author:	Jiri Suchomel <jsuchome@suse.cz>
#  $Id$
module Yast
  class ReadLdapConfEntryClient < Client
    def main
      Yast.include self, "testsuite.rb"
      # testedfiles: Ldap.ycp

      @READ = {
        "etc"     => {
          "ldap_conf" => {
            "v" => {
              "/etc/ldap.conf" => {
                "host"            => nil,
                "base"            => "dc=suse,dc=cz",
                "nss_base_passwd" => nil
              }
            }
          }
        },
        "product" => { "features" => { "EVMS_CONFIG" => "nazdar" } },
        "target"  => { "size" => -1 }
      }

      TESTSUITE_INIT([@READ, {}, {}], nil)

      Yast.import "Ldap"

      TEST(lambda { Ldap.ReadLdapConfEntry("host", "localhost") }, [
        @READ,
        {},
        {}
      ], 0)

      TEST(lambda { Ldap.ReadLdapConfEntry("base", "dc=test") }, [@READ, {}, {}], 0)

      TEST(lambda { Ldap.ReadLdapConfEntry("nss_base_passwd", "") }, [
        @READ,
        {},
        {}
      ], 0)

      nil
    end
  end
end

Yast::ReadLdapConfEntryClient.new.main
