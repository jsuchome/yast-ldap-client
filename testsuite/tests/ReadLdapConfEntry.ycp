/**
 *  ReadLdapConfEntry.ycp
 *  Test of Ldap::ReadLdapConfEntry function
 *  Jiri Suchomel <jsuchome@suse.cz>
 *  $Id$
 */

{

    include "testsuite.ycp";
    import "Ldap";

    map READ = $[
	"etc": $[
	    "ldap_conf" : $[
		"v" : $[
		    "/etc/ldap.conf": $[
			"host" : nil,
			"base" : "dc=suse,dc=cz",
			"nss_base_passwd" : nil,
		    ]
		]
	    ],
	]
    ];

    TEST(``(Ldap::ReadLdapConfEntry ("host", "localhost")), [ READ, [], [] ],0);

    TEST(``(Ldap::ReadLdapConfEntry ("base", "dc=test")), [ READ, [], [] ],0);

    TEST(``(Ldap::ReadLdapConfEntry ("nss_base_passwd", "")), [ READ,[],[]],0);
}
