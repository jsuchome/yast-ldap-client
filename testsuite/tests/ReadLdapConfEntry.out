Read	.etc.ldap_conf.v."/etc/ldap.conf"."host" nil
Return	localhost
Read	.etc.ldap_conf.v."/etc/ldap.conf"."base" "dc=suse,dc=cz"
Return	dc=suse,dc=cz
Read	.etc.ldap_conf.v."/etc/ldap.conf"."nss_base_passwd" nil
Return	
